### PR TITLE
Mapstore2 Elevation Slider tool

### PR DESCRIPTION
**Pull request by mricca**
**Layer configuration example:**

	"params": {
		"DIM_DEPTH": ["1.5"] 
	},
	"elevations": {
		"name": "DIM_DEPTH", _Name of the elevation params in geoserver_
		"units": "Meters", _Units of Measure_
		"positive": false, _to be set according to the meaning of the variable_
		"showChart": true, _Set to false if you do not view the chart with the slider_
		"values": ["1.5","5.0",...] _hard coded value from the capabilities_
	}

**Full example:**
```
{
	"group": "METOC Forecasts",
	"name": "gliders:salinity",
	"title": "Salinity",
	"type": "wms",
	"url": "http://cloudsdi.geo-solutions.it/geoserver/wms",
	"bbox": {
		"crs": "EPSG:4326",
			"bounds": {
			"minx": "6",
			"miny": "41",
			"maxx": "13",
			"maxy": "46"
		}
	},
	"visibility": true,
	"singleTile": true,
	"format": "image/png8",
	"params": {
		"DIM_DEPTH": ["1.5"]
	},
	"elevations": {
		"name": "DIM_DEPTH",
		"units": "Meters",
		"positive": false,
		"showChart": true,
		"values": ["1.5","5.0","10.0","15.0","20.0","25.0","30.0","35.0","40.0","45.0","50.0","60.0","70.0","80.0","90.0","100.0","120.0","140.0","160.0","180.0","200.0","225.0","250.0","275.0","300.0","350.0","400.0","450.0","500.0","600.0","700.0","800.0","900.0","1000.0","1200.0","1400.0","1600.0","1800.0","2000.0","2400.0","2800.0"]
	}
}
```

### CODE REVIEW

File: web/client/translations/data.de-DE [hunk 1]
```diff
@@ -53,6 +54,7 @@
         "save": "Speichern",
         "saveAs": "Speichern als...",
         "opacity": "Deckkraft",
+        "elevation": "Elevation",
         "close": "Schliessen",
```
Comment (MV88) [inline web/client/translations/data.de-DE]: should be translated this?

File: web/client/components/TOC/fragments/settings/Elevation.jsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                {this.renderElevationsChart(elevations)}
+                <div>
+                    <div key="elevation">
+                        {this.renderElevationsSlider(elevations)}
+                    </div>
+                </div>
+            </div>
+        );
+    },
+    calcolateRange(values, min, max, dif, firstVal, lastVal) {
```
Comment (mbarto) [inline web/client/components/TOC/fragments/settings/Elevation.jsx]: calculate

File: web/client/components/TOC/fragments/settings/Elevation.jsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        let arr = [];
+        let percText = "";
+        let range = {min: firstVal, max: lastVal};
+        values.map(function(currentValue, i) {
```
Comment (mbarto) [inline web/client/components/TOC/fragments/settings/Elevation.jsx]: map requires a pure function, if you have side effects use forEach

File: web/client/components/TOC/fragments/settings/Elevation.jsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        const start = this.props.element &&
+                        this.props.element.params &&
+                        this.props.element.params[this.props.elevations.name][0] || values[0];
+        const elevationName = {};
+        return (
+            <div id="mapstore-elevation">
+                <Slider
+                    snap= {true}
+                    start={[parseFloat((start))] || [0.0]}
+                    range= {this.calcolateRange(values, min, max, dif, firstVal, lastVal)}
```
Comment (mbarto) [inline web/client/components/TOC/fragments/settings/Elevation.jsx]: calculate

File: package.json [hunk 5]
```diff
@@ -161,6 +162,7 @@
     "url": "0.10.3",
     "uuid": "3.0.1",
     "w3c-schemas": "1.3.1",
+    "wnumb": "1.1.0",
```
Comment (mbarto) [inline package.json]: we already have I18N.NumberFormat for number formatting

**Pull request merged**